src/store/modules/blocks/actions.js

```diff
@@ -142,9 +142,9 @@ export default {
 
     const generation = await getGenerationFromHeightWrapper(height, this.state.epochUrl)
 
-    if (isEqual(state.block, generation)) {
+    if (isEqual(state.generation, generation)) {
```

Comment:
I think we just shouldn't download block with the same height twice because blockchain is immutable.